artifactory_test.go

```diff
@@ -5152,3 +5152,65 @@ func readerCloseAndAssert(t *testing.T, reader *content.ContentReader) {
 func readerGetErrorAndAssert(t *testing.T, reader *content.ContentReader) {
 	assert.NoError(t, reader.GetError(), "Couldn't get reader error")
 }
+
+// Terraform tests : can not be tested yet.
+
+func TestTerraformPublish(t *testing.T) {
+	initArtifactoryTest(t)
+	defer cleanArtifactoryTest()
+	createJfrogHomeConfig(t, true)
+	projectPath := prepareTerraformProject("terraformproject", t, true)
+	log.Info("project path : ", projectPath)
+	// Change working directory to be the project's local root.
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	chdirCallback := clientTestUtils.ChangeDirWithCallback(t, wd, filepath.Join(projectPath, "aws"))
+	defer chdirCallback()
+	artifactoryCli.SetPrefix("jf")
+
+	// Terraform publish
+	err = artifactoryCli.Exec("terraform", "publish", "--namespace=namespace", "--provider=provider", "--tag=tag", "--exclusions=*test*")
+	assert.NoError(t, err)
+	artifactoryCli.SetPrefix("jf rt")
+
+	// Download modules to 'result' directory.
+	chdirCallback()
+	assert.NoError(t, os.MkdirAll(tests.Out+"/results/", 0777))
+	runRt(t, "download", tests.TerraformRepo+"/namespace/provider/*", tests.Out+"/results/", "--explode=true")
+	// Validate
+	paths, err := fileutils.ListFilesRecursiveWalkIntoDirSymlink(tests.Out+"/results", false)
+	assert.NoError(t, err)
+	assert.NoError(t, tests.ValidateListsIdentical(tests.GetTerraformModulesFilesDownload(), paths))
+}
+
+func prepareTerraformProject(projectName string, t *testing.T, copyDirs bool) string {
+	projectPath := filepath.Join(tests.GetTestResourcesPath(), "terraform", projectName)
+	log.Info("project path : ", projectPath)
+	testdataTarget := filepath.Join(tests.Out, "terraformProject")
+	assert.NoError(t, os.MkdirAll(testdataTarget+string(os.PathSeparator), 0777))
+	// Copy terraform tests to test environment, so we can change project's config file.
+	wd, err := os.Getwd()
+	log.Info("pwd : ", wd)
+	log.Info("copy : ", projectPath, " to: ", testdataTarget)
+	assert.NoError(t, fileutils.CopyDir(projectPath, testdataTarget, copyDirs, nil))
+	os.Chdir(testdataTarget)
+	fmt.Println("list dir: ***************")
```

Comment:
Let's remove this log line, because I assume we don't want to see it every time the tests run.

I assume this test logging should go through cleanup, so I'm stopping here.